tests/error/util.py

```diff
@@ -12,11 +12,9 @@
 
 def run_error_test(file, capsys):
     file = pathlib.Path(file)
-    spec = importlib.util.spec_from_file_location("test_module", file)
-    py_module = importlib.util.module_from_spec(spec)
 
     with pytest.raises(GuppyError):
-        spec.loader.exec_module(py_module)
+        importlib.import_module(f"tests.error.{file.parent.name}.{file.name}")
```

Comment:
For some reason, inspect.getsource() doesn't work for classes that are loaded via importlib.util.module_from_spec(). Switching to importlib.import_module() seems to work though